internal/generator/ecc/template/point.go.tmpl

```diff
@@ -434,17 +434,33 @@ func (p *{{ $TJacobian }}) FromAffine(Q *{{ $TAffine }}) *{{ $TJacobian }} {
 
 // IsOnCurve returns true if p in on the curve
 func (p *{{ $TJacobian }}) IsOnCurve() bool {
-	var left, right, tmp  {{.CoordType}}
+	var left, right, tmp, ZZ  {{.CoordType}}
 	left.Square(&p.Y)
 	right.Square(&p.X).Mul(&right, &p.X)
-	tmp.Square(&p.Z).
-		Square(&tmp).
-		Mul(&tmp, &p.Z).
-		Mul(&tmp, &p.Z).
+	ZZ.Square(&p.Z)
+    tmp.Square(&ZZ).
+		Mul(&tmp, &ZZ)
 		{{- if eq .PointName "g1"}}
-			Mul(&tmp, &bCurveCoeff)
+            {{- if or (eq .Name "bls12-381") (eq .Name "bls24-317") (eq .Name "bw6-633")}}
+                tmp.Double(&tmp).Double(&tmp)
```

Comment:
+ for readability, may be nice to have a inline comment for each curve on "// mu by bCurveCoeff, which is == n"